src/errors/ApiError.ts

```diff
@@ -1,13 +1,31 @@
 import { AxiosResponse } from 'axios';
 
-import { MollieApiErrorLinks, Url } from '../data/global';
+import { idempotencyHeaderName } from '../communication/makeRetrying';
+import { Links, Url } from '../data/global';
 import Maybe from '../types/Maybe';
 
+type MollieApiErrorLinks = Record<string, Url> & Links;
+type MetaInfo = {
+  field?: string;
+  statusCode?: number;
+  idempotencyKey?: string;
+  title?: string;
+  links?: MollieApiErrorLinks;
+};
+
 export default class ApiError extends Error {
-  public constructor(message: string, protected title?: string, public readonly statusCode?: number, public readonly field?: string, protected links?: MollieApiErrorLinks) {
+  // Set the name to ApiError.
+  public readonly name: string = 'ApiError';
```

Comment:
`Error` actually comes with a `name` property [out of the box](//tc39.es/ecma262/#sec-error.prototype.name). I think we should use that instead of adding our own.

Reply:
discussed:
+ easier to set `Symbol.toStringTag`
- subtle usage differences

=> Decision to keep it like this.